ha/core/system_health/system_health.py

```diff
@@ -439,6 +439,9 @@ def process_event(self, healthevent: HealthEvent):
             if current_health:
                 current_health_dict = json.loads(current_health)
                 specific_info = current_health_dict["events"][0]["specific_info"]
+                # Get specific_info data from consul
+                if specific_info and specific_info.get('functional_type') and not healthevent.specific_info.get('functional_type'):
```

Comment:
HA should never expect functional_type to be present in the incoming alert. It should always be fetched from consul

Reply:
I think specific_info here is coming from consul only (its the specific info of already stored health in the consul).
Also, @gauri-bhosale I think, the last statement **not healthevent.specific_info.get('functional_type')** is not required here.